glean-core/rlb/src/private/labeled.rs

```diff
@@ -0,0 +1,389 @@
+            },
+            None,
+        );
+
+        // Record in 20 labels: it will go over the maximum number of supported
+        // dynamic labels.
+        for i in 0..=20 {
+            metric.get(format!("label_{}", i).as_str()).add(1);
+        }
+        // Record in a label once again.
+        metric.get("label_0").add(1);
+
+        // Initialize Glean.
+        let _ = new_glean(None, false);
+
+        assert_eq!(2, metric.get("label_0").test_get_value(None).unwrap());
+        for i in 1..15 {
+            assert_eq!(
+                1,
+                metric
+                    .get(format!("label_{}", i).as_str())
+                    .test_get_value(None)
+                    .unwrap()
+            );
+        }
+        assert_eq!(5, metric.get("__other__").test_get_value(None).unwrap());
+    }
+
+    #[test]
+    fn test_labeled_string_type() {
+        let _lock = lock_test();
+
+        let _ = new_glean(None, true);
+
+        let metric: LabeledMetric<StringMetric> = LabeledMetric::new(
+            CommonMetricData {
+                name: "labeled_string".into(),
+                category: "labeled".into(),
+                send_in_pings: vec!["test1".into()],
+                disabled: false,
+                ..Default::default()
+            },
+            None,
+        );
+
+        metric.get("label1").set("foo");
+        metric.get("label2").set("bar");
+        assert_eq!("foo", metric.get("label1").test_get_value("test1").unwrap());
+        assert_eq!("bar", metric.get("label2").test_get_value("test1").unwrap());
+    }
+
+    #[test]
+    fn test_labeled_boolean_type() {
+        let _lock = lock_test();
+
+        let _ = new_glean(None, true);
+
+        let metric: LabeledMetric<BooleanMetric> = LabeledMetric::new(
+            CommonMetricData {
+                name: "labeled_boolean".into(),
+                category: "labeled".into(),
+                send_in_pings: vec!["test1".into()],
+                disabled: false,
+                ..Default::default()
+            },
+            None,
+        );
+
+        metric.get("label1").set(false);
+        metric.get("label2").set(true);
+        assert!(!metric.get("label1").test_get_value("test1").unwrap());
+        assert!(metric.get("label2").test_get_value("test1").unwrap());
+    }
+
+    #[test]
+    fn test_invalid_labels_record_errors() {
+        let _lock = lock_test();
+
+        let _ = new_glean(None, true);
+
+        let metric: LabeledMetric<BooleanMetric> = LabeledMetric::new(
+            CommonMetricData {
+                name: "labeled_boolean".into(),
+                category: "labeled".into(),
+                send_in_pings: vec!["test1".into()],
+                disabled: false,
+                ..Default::default()
+            },
+            None,
+        );
+
+        let invalid_label = "!#I'm invalid#--_";
+        metric.get(invalid_label).set(true);
+        // TODO: Uncomment after bug 1677807 is fixed.
```

Comment:
Logging an error AND recording to the `__other__` label is kinda the behavior we stuck with ever since, so I doubt that the bug gets fixed in a way to make the below line pass.

Maybe make the assert explicitly check the `__other__` label to signal that the data _does_ get recorded?